components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -355,48 +356,30 @@ class GeckoEngineSession(
 
             val response = settings.requestInterceptor?.onLoadRequest(
                 this@GeckoEngineSession,
-                request.uri
+                request.uri,
+                request.hasUserGesture,
+                this@GeckoEngineSession.currentUrl?.tryGetHostFromUrl() == request.uri.tryGetHostFromUrl()
```

Comment:
nit: I would assign this to a `val isSameDomain` to make it easier to read here. 

I also realizes now that your original solution to pass both urls to the interceptor was easier to read, but just a nit. :)

Reply:
Agreed.  I'll update.